bundle/edu.gemini.pit/src/main/scala/edu/gemini/pit/ui/robot/ProblemRobot.scala

```diff
@@ -433,29 +432,26 @@ class ProblemRobot(s: ShellAdvisor) extends Robot {
     }
 
     private lazy val missingObsElementCheck = {
-      def fix[A](b: Band, g: ObsListGrouping[A]) {
-        s.inObsListView(b, _.Fixes.fixEmpty(g))
-      }
-
-      def check[A](s: String, g: ObsListGrouping[A]) =
-        p.observations.filter(g.lens.get(_).isEmpty) match {
+      def check[A](msg: String, g: ObsListGrouping[A], adder: ObsListView => Unit) =
+        //p.observations.filter(o => o.nonEmpty && g.lens.get(o).isEmpty) match {
+        p.observations.filter(o => g.lens.get(o).isEmpty) match {
           case Nil => None
-          case h :: Nil => Some(new Problem(Severity.Error, s"One observation has no $s.", "Observations", fix(h.band, g)))
-          case h :: tail => Some(new Problem(Severity.Error, s"${1 + tail.length} observations have no $s.", "Observations", fix(h.band, g)))
+          case o :: Nil => Some(new Problem(Severity.Error, s"One observation has no $msg.", "Observations", s.inObsListView(o.band, adder)))
```

Comment:
We now have an `adder` param to fix missing details using an editor that adds them instead of simply highlighting them, as per Bryan's request.

Reply:
i'm not sure what's the role of `adder`